cedar-lean/GUIDE.md

```diff
@@ -0,0 +1,62 @@
+# Style guidelines for Cedar Lean
+
+The Cedar Lean formalization uses the following coding conventions, adapted from [Lean 4](https://leanprover.github.io/lean4/doc/lean3changes.html#style-changes) and [Mathlib 4](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention).
+
+## Imports
+
+Sort `import` and `open` statements lexicographically. Keep imports to a minimum.
+
+## Casing
+
+* Type variables are lower-case Greek letters.
+* Theorem names and `Prop` terms are in `lower_snake_case`.
+* Inductive types, structures, typeclasses, modules, and namespaces use `UpperCamelCase`.
+* Constructors of inductive types use `lowerCamelCase`.
+* Function names are `lowerCamelCase` unless they return a `Prop`, in which case they are `UpperCamelCase`.
+* Everything else (e.g., structure fields and local variables) is `lowerCamelCase`.
+
+## Indentation
```

Comment:
Maybe we want to use the Lean pretty-printer to format our code? (https://github.com/leanprover/lean4/issues/369)

Reply:
Yes, it would be great to have the code auto-formatted.  My understanding is that the existing pretty printer is not quite ready for production yet.